pkg/health/heartbeat/heartbeat.go

```diff
@@ -175,28 +175,35 @@ func UpdateFullSnapshotLease(ctx context.Context, logger *logrus.Entry, fullSnap
 			}
 		}
 
-		// Do not update revision number if revision in existing Lease is already higher.
+		var rev int64
+		logString := "Renewed full snapshot lease"
+
+		// Do not renew lease if revision in existing Lease is already higher.
 		if fullSnapLease.Spec.HolderIdentity != nil {
-			rev, err := strconv.ParseInt(*fullSnapLease.Spec.HolderIdentity, 10, 64)
+			rev, err = strconv.ParseInt(*fullSnapLease.Spec.HolderIdentity, 10, 64)
 			if err != nil {
 				return err
 			}
-			if rev >= fullSnapshot.LastRevision {
+			if rev > fullSnapshot.LastRevision {
 				return nil
 			}
 		}
 
 		renewedLease := fullSnapLease.DeepCopy()
-		actor := strconv.FormatInt(fullSnapshot.LastRevision, 10)
-		renewedLease.Spec.HolderIdentity = &actor
 		renewedTime := time.Now()
 		renewedLease.Spec.RenewTime = &metav1.MicroTime{Time: renewedTime}
+		//Update revision number only if revisions in existing lease is lower
```

Comment:
It is not updating the revision , it is updating the lease spec HolderIdentity. 
Can you update this comment.

Reply:
Addressed all comments. PTAL